sirepo/template/madx.py

```diff
@@ -135,38 +131,22 @@ def background_percent_complete(report, run_dir, is_running):
 
 
 def get_application_data(data, **kwargs):
-    assert 'method' in data
-    assert data.method in _METHODS, \
-        'unknown application data method: {}'.format(data.method)
     if data.method == 'calculate_bunch_parameters':
         return _calc_bunch_parameters(data.bunch, data.command_beam, data.variables)
-    #TODO(pjm): move to template_command and share with elegant and opal
-    cv = _code_var(data.variables)
-    if data.method == 'rpn_value':
-        # accept array of values enclosed in curly braces
-        if re.search(r'^\{.*\}$', data.value):
-            data.result = ''
-            return data
-        v, err = cv.eval_var(data.value)
-        if err:
-            data.error = err
-        else:
-            data.result = v
-        return data
-    if data.method == 'recompute_rpn_cache_values':
-        cv.recompute_cache(data.cache)
-        return data
-    if data.method == 'validate_rpn_delete':
-        model_data = simulation_db.read_json(
-            simulation_db.sim_data_file(data.simulationType, data.simulationId))
-        data.error = cv.validate_var_delete(
-            data.name,
-            model_data,
-            _SCHEMA
-        )
+    if _code_var(data.variables).get_application_data(data, _SCHEMA, ignore_array_values=True):
         return data
```

Comment:
Missing a return statement when get_application_data returns false (which I don't think it should).

Reply:
It used to have assertion failure in all the templates, so I moved it to server.py - it will now raise an assertion if the get_application_data() did not produce output.

Reply:
Fast fail would suggest to fail in the common code as well as if no output was produced. If you throw the exception in side code_variables.py, you'll get the stack of what happened. If you throw it outside, you have no idea why it wasn't produced. Is there a case where checking the return value should not result in an exception?